server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts

```diff
@@ -107,14 +120,43 @@ export function create(
 
             handleResponse(Promise.all([createP, tenantKeyP]).then(([_, key]) => {
```

Comment:
This was not introduced by you, but we have an opportunity here to improve. If `generateToken === false`, we do not need to fetch the tenant key from Riddler here, so we should save resources by not fetching. Try something similar to my previous suggestion:

```ts
const enableDiscovery: boolean = request.body.enableDiscovery ?? false;

// Handle backwards compatibility for older driver versions.
// TODO: remove condition once old drivers are phased out and all clients can handle object response
const clientAcceptsObjectResponse = enableDiscovery !== undefined || generateToken === true;
if (clientAcceptsObjectResponse) {
  const responseBody = { id, token: undefined, session: undefined };
  if (generateToken) {
    // Generate creation token given a jwt from header
    const authorizationHeader = request.header("Authorization");
    const tokenRegex = /Basic (.+)/;
    const tokenMatch = tokenRegex.exec(authorizationHeader);
    const token = tokenMatch[1];
    const tenantKey = await tenantManager.getKey(tenantId);
    responseBody.token = getCreationToken(getCreationToken(token, key, id));
  }
  if (enableDiscovery) {
    const session: ISession = {
       ordererUrl,
       historianUrl,
       isSessionAlive: false,
     };
     responseBody.session = session;
  }
  handleResponse(createP.then(() => responseBody), response, undefined, 201);
} else {
  handleResponse(createP.then(() => id), response, undefined, 201);
}
```

This gives the benefit of (1) never doing more work than needed, and (2) if more conditional response properties are added then we don't have to maintain a complete if..elseif matrix for all possibilities.

Reply:
Yeah, exactly! Just refactor the code the same as the above logic.